database/migrations/2017_04_27_141456_create_submission_states_table.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateSubmissionStatesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('submission_states', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('transition');
+            $table->string('from')->nullable(); // What we were before this event.
+            $table->integer('user_id')->unsigned(); // a User who made this happen.
+            $table->integer('submission_id')->unsigned();
+            $table->string('to');   // What state we are now.
```

Comment:
It's a record of a transition with a from and to state. Yes, the current state is fetched *from* the top record in a voucher's state stack, and is on a field called voucher->currentstate.